modules/calc/src/main/java/com/opengamma/strata/calc/runner/DerivedCalculationFunctionWrapper.java

```diff
@@ -0,0 +1,194 @@
+
+  /**
+   * Creates a new function which invokes the delegate function, passes the result to the derived function
+   * and returns the combined results.
+   *
+   * @param calculationFunction  a function which calculates one measure using the measure values calculated by the other function
+   * @param delegate  a function which calculates multiple measures
+   */
+  DerivedCalculationFunctionWrapper(
+      DerivedCalculationFunction<T, R> calculationFunction,
+      CalculationFunction<T> delegate) {
+
+    this.calculationFunction = calculationFunction;
+    this.delegate = delegate;
+
+    Set<Measure> delegateMeasures = delegate.supportedMeasures();
+    this.requiredMeasuresSupported = delegateMeasures.containsAll(calculationFunction.requiredMeasures());
+    this.supportedMeasures = requiredMeasuresSupported ?
+        ImmutableSet.<Measure>builder().addAll(delegateMeasures).add(calculationFunction.measure()).build() :
+        delegateMeasures;
+  }
+
+  @Override
+  public Class<T> targetType() {
+    return calculationFunction.targetType();
+  }
+
+  @Override
+  public Set<Measure> supportedMeasures() {
+    return supportedMeasures;
+  }
+
+  @Override
+  public Currency naturalCurrency(T target, ReferenceData refData) {
+    return delegate.naturalCurrency(target, refData);
+  }
+
+  @Override
+  public FunctionRequirements requirements(
+      T target,
+      Set<Measure> measures,
+      CalculationParameters parameters,
+      ReferenceData refData) {
+
+    FunctionRequirements delegateRequirements = delegate.requirements(target, measures, parameters, refData);
+    FunctionRequirements functionRequirements = calculationFunction.requirements(target, parameters, refData);
+    return delegateRequirements.combinedWith(functionRequirements);
+  }
+
+  @Override
+  public Map<Measure, Result<?>> calculate(
+      T target,
+      Set<Measure> measures,
+      CalculationParameters parameters,
+      ScenarioMarketData marketData,
+      ReferenceData refData) {
+
+    // Does the caller want the extra measure?
+    boolean measureRequested = measures.contains(calculationFunction.measure());
+
+    // We need to add the measures required to calculate the measure
+    // But we don't need to do that if the delegate can't provide them or if the user hasn't asked for the measure
+    Set<Measure> requiredMeasures = requiredMeasuresSupported && measureRequested ?
+        Sets.union(measures, calculationFunction.requiredMeasures()) :
+        measures;
+
+    Map<Measure, Result<?>> delegateResults = delegate.calculate(target, requiredMeasures, parameters, marketData, refData);
+
+    // The user didn't ask for the extra measure so just return the measures calculated by the delegate
+    if (!measureRequested) {
+      return delegateResults;
+    }
+    // Calculate the extra measure provided by calculationFunction
+    Result<?> result = calculateMeasure(target, delegateResults, parameters, marketData, refData);
```

Comment:
Should the delegate receive `Map<Measure, Result<?>>`, or could it be `Map<Measure, Object>`. This would avoid exposing failures to the delegate, but does remove the delegates ability to handle failures. Not sure.